File: one.py
#!/usr/bin/python 
import socket
import sys
import os

def main(argv=None):
	if argv is None:
		argv = sys.argv
	input = "%d\007" % os.getuid();
	i = 0;
	for arg in argv:
		if i == 0:
			i = 1;
		else:
			input += arg + " "
	#create a UNIX, STREAMing socket
	s = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
	# connect to socket
	s.connect("/tmp/shelly")
	s.send(input)
	input = s.recv(4096)
	print input 
	#close socket
	s.close()

if __name__ == "__main__":
	main()

File: one.pl
#!/usr/bin/perl -w

use strict;
use warnings;
use IO::Socket;

my $socket = "/tmp/shelly";

my $client = IO::Socket::UNIX->new(Peer => $socket,
                                   Type => SOCK_STREAM) or die $@;

my $buf = $ENV{USER} . "\007" . join(" ", @ARGV);

$client->send($buf, 1024) or die $@;

$client->recv($buf, 1024) or die $@;

print $buf . "\n";

